managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/EntryCacheImpl.java

```diff
@@ -208,7 +208,12 @@ private void asyncReadEntry0(ReadHandle lh, PositionImpl position, final ReadEnt
             EntryImpl cachedEntry = EntryImpl.create(entry);
             entry.release();
             manager.mlFactoryMBean.recordCacheHit(cachedEntry.getLength());
-            callback.readEntryComplete(cachedEntry, ctx);
+            EntryCacheCounter entryCacheCounter = EntryCacheCounter.EntryCacheCounterBuilder
+                    .getBuilder()
+                    .setCacheHitCount(1)
+                    .setCacheHitSize(cachedEntry.getLength())
+                    .build();
```

Comment:
This will create more objects here which will increase the jvm GC workload. It's better to use a callback here to record the metrics in the subscription.